testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/principaltransformers/CasePrincipalTransformerTestCase.java

```diff
@@ -0,0 +1,181 @@
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.as.test.integration.security.common.servlets.SimpleSecuredServlet;
+import org.jboss.as.test.integration.security.common.servlets.SimpleServlet;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.security.common.elytron.PropertyFileBasedDomain;
+
+
+/**
+ * Test case for 'case-principal-transformer' Elytron subsystem resource.
+ *
+ * @author Sonia Zaldana Calles <szaldana@redhat.com>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({CasePrincipalTransformerTestCase.SetupTask.class})
+public class CasePrincipalTransformerTestCase {
+
+    private static final String DEP_SECURITY_DOMAIN = "case-principal-domain";
+    private static final String USER = "USER1";
+    private static final String PASSWORD = "password1";
+    private static final String ROLE = "JBossAdmin";
+    @ArquillianResource
+    private static ManagementClient client;
+
+    @Deployment(name = DEP_SECURITY_DOMAIN)
+    public static WebArchive createDeployment() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, DEP_SECURITY_DOMAIN + ".war");
+        war.addClasses(SimpleServlet.class);
+        war.addClasses(SimpleSecuredServlet.class);
+        war.addAsWebInfResource(ConstantPrincipalTransformerTestCase.class.getPackage(), "principal-transformer-web.xml", "web.xml");
+        war.addAsWebInfResource(Utils.getJBossWebXmlAsset(DEP_SECURITY_DOMAIN), "jboss-web.xml");
+        return war;
+    }
+
+    /**
+     * Test whether name of user passed to security domain is adjusted into upper case with
+     * the 'case-principal-transformer'. Test also checks that authentication passes for correct password.
+     */
+    @Test
+    @OperateOnDeployment(DEP_SECURITY_DOMAIN)
+    public void testPassingExistingUser(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareUrl(webAppURL);
+        Utils.makeCallWithBasicAuthn(url, "user1", PASSWORD, SC_OK); // "USER1" is stored in the realm
+    }
+
+    /**
+     * Test checks that authentication fails for incorrect password and incorrect user if 'case-principal-transformer'
+     * is used.
+     */
+    @Test
+    @OperateOnDeployment(DEP_SECURITY_DOMAIN)
+    public void testPassingNonexistentUser(@ArquillianResource URL webAppURL) throws Exception {
+        URL url = prepareUrl(webAppURL);
+        Utils.makeCallWithBasicAuthn(url, "wrongUser", PASSWORD, SC_UNAUTHORIZED);
+    }
+
+    /**
+     * Tests whether authentication fails when the principal transformer is not configured
+     */
+    @Test
+    @OperateOnDeployment(DEP_SECURITY_DOMAIN)
+    public void testPassingExistingUserWithoutPrincipalTransformer(@ArquillianResource URL webAppURL) throws Exception {
+        SetupTask.removePrincipalTransformer(client);
```

Comment:
Instead of removing the principal transformer, you could make use of a second security domain that doesn't have a principal transformer configured

Reply:
I pushed a new commit where I used different domains to tackle the cases where the principal transformer is configured to adjust to lower case and a domain with no principal transformer. I pushed this as a separate commit just so in case I need to roll back to the previous state. Let me know if the new commit incorporates the feedback appropriately :)